diff -Nru vdr-1.7.11-vanilla/pat.c vdr-1.7.11-disable_ca_updates/pat.c
--- vdr-1.7.11-vanilla/pat.c	2010-01-06 17:06:01.000000000 +0200
+++ vdr-1.7.11-disable_ca_updates/pat.c	2010-01-06 17:12:42.000000000 +0200
@@ -458,6 +458,7 @@
             }
         if (Setup.UpdateChannels >= 2) {
            Channel->SetPids(Vpid, Ppid, Vtype, Apids, ALangs, Dpids, DLangs, Spids, SLangs, Tpid);
+           if (!Channel->IsPlug())
            Channel->SetCaIds(CaDescriptors->CaIds());
            Channel->SetSubtitlingDescriptors(SubtitlingTypes, CompositionPageIds, AncillaryPageIds);
            }
