diff -Nru vdr-1.7.8-vanilla/pat.c vdr-1.7.8-disable_ca_updates/pat.c
--- vdr-1.7.8-vanilla/pat.c	2009-06-15 17:31:45.000000000 +0300
+++ vdr-1.7.8-disable_ca_updates/pat.c	2009-06-17 15:09:13.000000000 +0300
@@ -444,6 +444,7 @@
             }
         if (Setup.UpdateChannels >= 2) {
            Channel->SetPids(Vpid, Ppid, Vtype, Apids, ALangs, Dpids, DLangs, Spids, SLangs, Tpid);
+           if (!Channel->IsPlug())
            Channel->SetCaIds(CaDescriptors->CaIds());
            }
         Channel->SetCaDescriptors(CaDescriptorHandler.AddCaDescriptors(CaDescriptors));
