diff -Nru vdr-1.6.0-vanilla/pat.c vdr-1.6.0-disable-ca-updates/pat.c
--- vdr-1.6.0-vanilla/pat.c	2008-02-08 15:48:31.000000000 +0200
+++ vdr-1.6.0-disable-ca-updates/pat.c	2009-03-07 14:56:42.000000000 +0200
@@ -440,6 +440,7 @@
             }
         if (Setup.UpdateChannels >= 2) {
            Channel->SetPids(Vpid, Vpid ? Ppid : 0, Apids, ALangs, Dpids, DLangs, Spids, SLangs, Tpid);
+           if (!Channel->IsPlug())
            Channel->SetCaIds(CaDescriptors->CaIds());
            }
         Channel->SetCaDescriptors(CaDescriptorHandler.AddCaDescriptors(CaDescriptors));
