diff -Nru vdr-1.7.14-vanilla/pat.c vdr-1.7.14-disable_ca_updates/pat.c
--- vdr-1.7.14-vanilla/pat.c	2010-03-14 16:57:06.000000000 +0200
+++ vdr-1.7.14-disable_ca_updates/pat.c	2010-03-14 17:35:55.000000000 +0200
@@ -481,6 +481,7 @@
             }
         if (Setup.UpdateChannels >= 2) {
            Channel->SetPids(Vpid, Ppid, Vtype, Apids, ALangs, Dpids, DLangs, Spids, SLangs, Tpid);
+           if (!cSource::IsType(Channel->Source(), 'I'))
            Channel->SetCaIds(CaDescriptors->CaIds());
            Channel->SetSubtitlingDescriptors(SubtitlingTypes, CompositionPageIds, AncillaryPageIds);
            }
