diff -Nru vdr-1.7.16-vanilla/pat.c vdr-1.7.16-disable_ca_updates/pat.c
--- vdr-1.7.16-vanilla/pat.c	2010-09-19 19:42:08.000000000 +0300
+++ vdr-1.7.16-disable_ca_updates/pat.c	2010-09-19 20:43:10.000000000 +0300
@@ -491,6 +491,7 @@
             }
         if (Setup.UpdateChannels >= 2) {
            Channel->SetPids(Vpid, Ppid, Vtype, Apids, Atypes, ALangs, Dpids, Dtypes, DLangs, Spids, SLangs, Tpid);
+           if (!cSource::IsType(Channel->Source(), 'I'))
            Channel->SetCaIds(CaDescriptors->CaIds());
            Channel->SetSubtitlingDescriptors(SubtitlingTypes, CompositionPageIds, AncillaryPageIds);
            }
