diff -Nru vdr-1.7.26-vanilla/pat.c vdr-1.7.26-disable_ca_updates/pat.c
--- vdr-1.7.26-vanilla/pat.c	2012-03-10 21:07:52.000000000 +0200
+++ vdr-1.7.26-disable_ca_updates/pat.c	2012-03-10 23:19:30.000000000 +0200
@@ -537,6 +537,7 @@
             }
         if (Setup.UpdateChannels >= 2) {
            Channel->SetPids(Vpid, Ppid, Vtype, Apids, Atypes, ALangs, Dpids, Dtypes, DLangs, Spids, SLangs, Tpid);
+           if (!cSource::IsType(Channel->Source(), 'I'))
            Channel->SetCaIds(CaDescriptors->CaIds());
            Channel->SetSubtitlingDescriptors(SubtitlingTypes, CompositionPageIds, AncillaryPageIds);
            }
