diff -Nru vdr-1.7.20-vanilla/pat.c vdr-1.7.20-disable_ca_updates/pat.c
--- vdr-1.7.20-vanilla/pat.c	2011-08-17 08:42:25.000000000 +0300
+++ vdr-1.7.20-disable_ca_updates/pat.c	2011-08-20 20:24:15.000000000 +0300
@@ -537,6 +537,7 @@
             }
         if (Setup.UpdateChannels >= 2) {
            Channel->SetPids(Vpid, Ppid, Vtype, Apids, Atypes, ALangs, Dpids, Dtypes, DLangs, Spids, SLangs, Tpid);
+           if (!cSource::IsType(Channel->Source(), 'I'))
            Channel->SetCaIds(CaDescriptors->CaIds());
            Channel->SetSubtitlingDescriptors(SubtitlingTypes, CompositionPageIds, AncillaryPageIds);
            }
