diff -Nru vdr-1.7.17-vanilla/eitscan.c vdr-1.7.17-disable_eitscan/eitscan.c
--- vdr-1.7.17-vanilla/eitscan.c	2011-03-13 15:06:48.000000000 +0200
+++ vdr-1.7.17-disable_eitscan/eitscan.c	2011-03-13 23:44:38.000000000 +0200
@@ -146,7 +146,7 @@
                if (Device) {
                   for (cScanData *ScanData = scanList->First(); ScanData; ScanData = scanList->Next(ScanData)) {
                       const cChannel *Channel = ScanData->GetChannel();
-                      if (Channel) {
+                      if (Channel && !cSource::IsType(Channel->Source(), 'I')) {
                          if (!Channel->Ca() || Channel->Ca() == Device->DeviceNumber() + 1 || Channel->Ca() >= CA_ENCRYPTED_MIN) {
                             if (Device->ProvidesTransponder(Channel)) {
                                if (!Device->Receiving()) {
